tests/unit/test_dataframe_polars.py

```diff
@@ -2443,12 +2443,36 @@ def test_join_different_table(
     assert_pandas_df_equal(bf_result, pd_result, ignore_order=True)
 
 
-def test_join_duplicate_columns_raises_not_implemented(scalars_dfs):
+@all_joins
+def test_join_raise_when_param_on_duplicate_with_column(scalars_df_index, how):
+    if how == "cross":
+        return
+    bf_df_a = scalars_df_index[["string_col", "int64_col"]].rename(
+        columns={"int64_col": "string_col"}
+    )
+    bf_df_a.index.name = "string_col"
+    bf_df_b = scalars_df_index.dropna()["string_col"]
+    with pytest.raises(ValueError):
```

Comment:
Please add a `match=` argument so we know the expected `ValueError` is reached, not a different one by accident. https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises